### ISSUE TITLE
Work with non json responses

### ISSUE THREAD

**Issue by dothebart** [arangodb/arangodb-java-driver#25]
In some cases the server will not reply with a json response; If thats the case the java driver seems to invoke the json driver anyways, which then in term throws an exception. 
The resulting exception however doesn't contain any useful information about the real error itself.

Thus the javadriver should check the content type, and catch parser errors so it can give the user the actual error message.

It seems as if this line is missing the exception handling:
at com.arangodb.entity.EntityFactory.createEntity(EntityFactory.java:109)

**Comment by maratkalibek** [comment]
how to reproduce that case? any testing data.

**Comment by dothebart** [comment]
```
telnet 127.0.0.1 8529
Trying 127.0.0.1...
Connected to 127.0.0.1.
Escape character is '^]'.
GET /blarg HTTP/1.0

HTTP/1.1 503 Service Unavailable
Server: ArangoDB
Connection: Close
Content-Type: text/html; charset=utf-8
Content-Length: 160

<html><head><title>Service Unavailable</title></head><body><p>This service is temporarily not available. Please check the log file for errors.</p></body></html>Connection closed by foreign host.
```
So if for some reason arangodb-java gets outside of the regular URL scheme, it may not get json, and therefore has to cope with it.

**Comment by a-brandt** [comment]
fixed in 2.7.3-SNAPSHOT
